mqtt/mqtt-broker/tests/smoke.rs

```diff
@@ -0,0 +1,114 @@
+use mqtt3::proto::QoS::{AtLeastOnce, AtMostOnce, ExactlyOnce};
+use mqtt3::proto::SubscribeTo;
+use mqtt3::{Event, ReceivedPublication};
+use mqtt_broker::{AuthId, BrokerBuilder};
+
+mod common;
+
+#[tokio::test]
+async fn basic_connect_clean_session() {
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let (broker_shutdown, broker_task, address) = common::start_server(broker);
+
+    let mut client = common::TestClientBuilder::default()
+        .address(address)
+        .client_id("mqtt-smoke-tests")
+        .build();
+
+    matches::assert_matches!(
+        client.events_receiver.recv().await,
+        Some(Event::NewConnection { reset_session }) if reset_session
+    );
+
+    broker_shutdown.send(()).expect("can't stop the broker");
+    broker_task
+        .await
+        .unwrap()
+        .expect("can't wait for the broker");
+}
+
+#[tokio::test]
+async fn basic_pub_sub() {
+    const TOPIC: &str = "topic/A";
+
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let (broker_shutdown, broker_task, address) = common::start_server(broker);
+
+    let mut client = common::TestClientBuilder::default()
```

Comment:
I know in mqtt3 crate author used fully qualified names, but in other code we should utilized usings